TestPeriodic fails: len(actions) = 0, expected >= 1

via semaphore,

```
2016-07-26 19:56:41.103215 N | etcdserver: Starting auto-compaction at revision 1
2016-07-26 19:56:41.103422 N | etcdserver: Finished auto-compaction at revision 1
2016-07-26 19:56:41.242566 N | etcdserver: Starting auto-compaction at revision 13
2016-07-26 19:56:41.242635 N | etcdserver: Finished auto-compaction at revision 13
2016-07-26 19:56:41.374956 N | etcdserver: Starting auto-compaction at revision 25
2016-07-26 19:56:41.375034 N | etcdserver: Finished auto-compaction at revision 25
2016-07-26 19:56:41.791289 N | etcdserver: Starting auto-compaction at revision 25
2016-07-26 19:56:41.791354 N | etcdserver: Finished auto-compaction at revision 25
2016-07-26 19:56:41.947325 N | etcdserver: Starting auto-compaction at revision 1
2016-07-26 19:56:41.947393 N | etcdserver: Finished auto-compaction at revision 1
2016-07-26 19:56:42.134529 N | etcdserver: Starting auto-compaction at revision 13
2016-07-26 19:56:42.134619 N | etcdserver: Finished auto-compaction at revision 13
2016-07-26 19:56:42.278163 N | etcdserver: Starting auto-compaction at revision 25
2016-07-26 19:56:42.278234 N | etcdserver: Finished auto-compaction at revision 25
2016-07-26 19:56:42.709166 N | etcdserver: Starting auto-compaction at revision 25
2016-07-26 19:56:42.709414 N | etcdserver: Finished auto-compaction at revision 25
--- FAIL: TestPeriodic (10.13s)
    compactor_test.go:51: len(actions) = 0, expected >= 1
2016-07-26 19:56:53.284981 N | etcdserver: Starting auto-compaction at revision 25
2016-07-26 19:56:53.285066 N | etcdserver: Finished auto-compaction at revision 25
FAIL
```

--- a/compactor/compactor_test.go
+++ b/compactor/compactor_test.go
@@ -40,12 +40,15 @@ func TestPeriodic(t *testing.T) {
 	defer tb.Stop()
 
 	n := int(time.Hour / checkCompactionInterval)
+	// collect 3 hours of revisions
 	for i := 0; i < 3; i++ {
+		// advance one hour, one revision for each interval
 		for j := 0; j < n; j++ {
-			rg.Wait(1)
 			fc.Advance(checkCompactionInterval)
+			rg.Wait(1)
 		}
-
+		// ready to acknowledge hour "i"; unblock clock
+		fc.Advance(checkCompactionInterval)
 		a, err := compactable.Wait(1)
 		if err != nil {
 			t.Fatal(err)

